app/controllers/catalog_controller.rb

```diff
@@ -38,6 +38,20 @@ def show
     end
   end
 
+  def index
+    query_param = params.fetch(:q)
```

Comment:
issue: This raises an error when the query param is empty.
Replacing with `params[:q]` fixes the issue.